dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -69,7 +74,7 @@ inline void ParseSampleDesc(std::vector<SampleDesc>& samples_container,
   // Reading consecutive components
   ComponentDesc component;
   while (components_stream >> component.ext) {
-    if (index_version == "v1.2") {
+    if (index_version == ParseIndexVersion("v1.2")) {
```

Comment:
Maybe `ParseIndexVersion("v1.2")` could be a static constant as it is bound to given DALI build.

Reply:
This code is terribly inefficient. It parses a constant version string over and over again in a loop.
We should either use:
- a numerical value obtained by calling a constexpr function like `MakeWDSIndexVersion(1, 2)`) - most favorable
- direct numerical value (e.g. 100200 or whatever our representation is) or a macro - not so good in C++, but works in C (if we ever expose this API anywhere)
- at the very least, storing `static const version_1_2 = ParseIndexVersion("v1.2")` - least favorable; we shouldn't need to resort to strings

Reply:
> This code is terribly inefficient. It parses a constant version string over and over again in a loop.

I believe it is resolved in compilation time, isn't it?

Reply:
No, it isn't!